### ISSUE TITLE
SSL problem with >= 4.1.52

### ISSUE THREAD

**Issue by imeszaros** [netty/netty#10720]
We're using Netty with [Jooby](https://github.com/jooby-project/jooby). We have a PKCS12 certstore to provide the cert-chain and private key. This worked fine with Jooby 2.9.0 (Netty 4.1.50), but with Jooby 2.9.1 (Netty 4.1.52) it fails like this when an HTTPS request is issued:

```
io.netty.handler.codec.DecoderException: javax.net.ssl.SSLProtocolException: No common named group
	at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:471)
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:276)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357)
	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919)
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:166)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:714)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysPlain(NioEventLoop.java:615)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:578)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:484)
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:989)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: javax.net.ssl.SSLProtocolException: No common named group
	at java.base/sun.security.ssl.Alert.createSSLException(Alert.java:126)
	at java.base/sun.security.ssl.TransportContext.fatal(TransportContext.java:321)
	at java.base/sun.security.ssl.TransportContext.fatal(TransportContext.java:264)
	at java.base/sun.security.ssl.TransportContext.fatal(TransportContext.java:259)
	at java.base/sun.security.ssl.KeyShareExtension$HRRKeyShareProducer.produce(KeyShareExtension.java:826)
	at java.base/sun.security.ssl.SSLExtension.produce(SSLExtension.java:532)
	at java.base/sun.security.ssl.SSLExtensions.produce(SSLExtensions.java:249)
	at java.base/sun.security.ssl.ServerHello$T13HelloRetryRequestProducer.produce(ServerHello.java:770)
	at java.base/sun.security.ssl.SSLHandshake.produce(SSLHandshake.java:436)
	at java.base/sun.security.ssl.ClientHello$T13ClientHelloConsumer.goHelloRetryRequest(ClientHello.java:1170)
	at java.base/sun.security.ssl.ClientHello$T13ClientHelloConsumer.consume(ClientHello.java:1158)
	at java.base/sun.security.ssl.ClientHello$ClientHelloConsumer.onClientHello(ClientHello.java:849)
	at java.base/sun.security.ssl.ClientHello$ClientHelloConsumer.consume(ClientHello.java:810)
	at java.base/sun.security.ssl.SSLHandshake.consume(SSLHandshake.java:392)
	at java.base/sun.security.ssl.HandshakeContext.dispatch(HandshakeContext.java:444)
	at java.base/sun.security.ssl.SSLEngineImpl$DelegatedTask$DelegatedAction.run(SSLEngineImpl.java:1065)
	at java.base/sun.security.ssl.SSLEngineImpl$DelegatedTask$DelegatedAction.run(SSLEngineImpl.java:1052)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/sun.security.ssl.SSLEngineImpl$DelegatedTask.run(SSLEngineImpl.java:999)
	at io.netty.handler.ssl.SslHandler.runAllDelegatedTasks(SslHandler.java:1557)
	at io.netty.handler.ssl.SslHandler.runDelegatedTasks(SslHandler.java:1571)
	at io.netty.handler.ssl.SslHandler.unwrap(SslHandler.java:1455)
	at io.netty.handler.ssl.SslHandler.decodeJdkCompatible(SslHandler.java:1282)
	at io.netty.handler.ssl.SslHandler.decode(SslHandler.java:1329)
	at io.netty.handler.codec.ByteToMessageDecoder.decodeRemovalReentryProtection(ByteToMessageDecoder.java:501)
	at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:440)
	... 17 common frames omitted
Caused by: java.io.IOException: No common named group
	... 39 common frames omitted
```

I checked and it works fine with Jooby 2.9.1 and with **Netty forcibly downgraded to 4.1.51** so this may be an issue introduced in  4.1.52.

We're using OpenJDK 11.0.2 on Windows.

**Comment by normanmaurer** [comment]
Can you show how you configure the SslContextBuilder ?

**Comment by imeszaros** [comment]
Jooby first creates a `javax.net.ssl.SSLContext` [here](https://github.com/jooby-project/jooby/blob/2.x/jooby/src/main/java/io/jooby/internal/SslPkcs12Provider.java), then before passing it to Netty it wraps it with `io.netty.handler.ssl.JdkSslContext` [here](https://github.com/jooby-project/jooby/blob/2f4d2ed6063595e482b8898111ceea933307e168/modules/jooby-netty/src/main/java/io/jooby/netty/Netty.java#L181). It seems that `SslContextBuilder` is not used. I hope this was what you were looking for.

**Comment by normanmaurer** [comment]
@imeszaros thanks a lot... can you share a reproducer ?

**Comment by imeszaros** [comment]
I could not reproduce the issue with the built in or a custom self-signed certificate. Maybe it depends on the contents of the PKCS12 file.

Our file contains the following:
- a root CA
- an intermediate CA
- a certificate issued for our domain
- private key for the certificate (2048 bit)

Which properties of the above should I check to be able to identify the issue?

**Comment by normanmaurer** [comment]
Can you try explicit setting the protocol to TLSv1.2 ?

**Comment by imeszaros** [comment]
@normanmaurer Your suggestion was really helpful, thanks.

I tried it with TLSv1.2 forced, and it worked. I did some research and it turned out that older JDK releases had some problems with TLS 1.3 (https://webtide.com/openjdk-11-and-tls-1-3-issues/, https://jira.atlassian.com/browse/BSERV-12374, https://stackoverflow.com/questions/60955206/java-11-sslhandshakeexception-with-tls-1-3-how-do-i-revert-to-tls-1-2) so I upgraded our environment to OpenJDK **11.0.8** and now the problem is gone.

Thanks for your help again.
